dataflux_core/download.py

```diff
@@ -396,6 +397,106 @@ def dataflux_download(
     return res
 
 
+def dataflux_download_lazy(
+    project_name: str,
+    bucket_name: str,
+    objects: list[tuple[str, int]],
+    storage_client: object = None,
+    dataflux_download_optimization_params: DataFluxDownloadOptimizationParams = None,
+    threading_enabled=False,
+) -> Iterator[bytes]:
+    """Perform the DataFlux download algorithm to download the object contents as bytes in a lazy fashion.
+
+    Args:
+        project_name: the name of the GCP project.
+        bucket_name: the name of the GCS bucket that holds the objects to compose.
+            The function uploads the the composed object to this bucket too.
+        objects: A list of tuples which indicate the object names and sizes (in bytes) in the bucket.
+            Example: [("object_name_A", 1000), ("object_name_B", 2000)]
+        storage_client: the google.cloud.storage.Client initialized with the project.
+            If not defined, the function will initialize the client with the project_name.
+        dataflux_download_optimization_params: the paramemters used to optimize the download performance.
+    Returns:
+        An iterator of the contents of the object in bytes.
+    """
+    if storage_client is None:
+        storage_client = storage.Client(project=project_name)
```

Comment:
Ideally the "storage_client" should be passed from "dataflux-pytorch" that has the user agent info but since users can change the code easily there I modified all places in `download.py` to add the user agent.